transit.go

```diff
@@ -308,6 +310,165 @@ func (t *Transit) DecryptBatch(key string, opts TransitDecryptOptionsBatch) (*Tr
+	res := &TransitSignResponseBatch{}
+
+	for i := range opts.BatchInput {
+		opts.BatchInput[i].Input = base64.StdEncoding.EncodeToString([]byte(opts.BatchInput[i].Input))
+	}
+
+	err := t.client.Write([]string{"v1", t.MountPoint, "sign", url.PathEscape(key)}, opts, res, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+type TransitVerifyOptions struct {
+	Input               string `json:"input"`
+	Signature           string `json:"signature"`
+	HashAlgorithm       string `json:"hash_algorithm,omitempty"`
+	Context             string `json:"context,omitempty"`
+	Prehashed           bool   `json:"prehashed,omitempty"`
+	SignatureAlgorithm  string `json:"signature_algorithm,omitempty"`
+	MarshalingAlgorithm string `json:"marshaling_algorithm,omitempty"`
+	SaltLength          string `json:"salt_length,omitempty"`
+}
+
+type TransitVerifyResponse struct {
+	Data struct {
+		Valid bool `json:"valid"`
+	} `json:"data"`
+}
+
+func (t *Transit) Verify(key string, opts *TransitVerifyOptions) (*TransitVerifyResponse, error) {
+	res := &TransitVerifyResponse{}
+
+	opts.Input = base64.StdEncoding.EncodeToString([]byte(opts.Input))
+
+	err := t.client.Write([]string{"v1", t.MountPoint, "verify", url.PathEscape(key)}, opts, res, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+type TransitBatchVerifyInput struct {
+	Input     string `json:"input"`
+	Signature string `json:"signature"`
+	Context   string `json:"context,omitempty"`
+}
+
+type TransitBatchVerifyData struct {
+	Valid bool `json:"valid"`
+}
+
+type TransitVerifyOptionsBatch struct {
+	BatchInput          []TransitBatchVerifyInput `json:"batch_input"`
+	HashAlgorithm       string                    `json:"hash_algorithm,omitempty"`
+	Context             string                    `json:"context,omitempty"`
+	Prehashed           bool                      `json:"prehashed,omitempty"`
+	SignatureAlgorithm  string                    `json:"signature_algorithm,omitempty"`
+	MarshalingAlgorithm string                    `json:"marshaling_algorithm,omitempty"`
+	SaltLength          string                    `json:"salt_length,omitempty"`
+}
+
+type TransitVerifyResponseBatch struct {
+	Data struct {
+		BatchResults []TransitBatchVerifyData `json:"batch_results"`
+	} `json:"data"`
+}
+
+func (t *Transit) VerifyBatch(key string, opts *TransitVerifyOptionsBatch) (*TransitVerifyResponseBatch, error) {
+	res := &TransitVerifyResponseBatch{}
+
+	for i := range opts.BatchInput {
+		opts.BatchInput[i].Input = base64.StdEncoding.EncodeToString([]byte(opts.BatchInput[i].Input))
+	}
+
+	err := t.client.Write([]string{"v1", t.MountPoint, "verify", url.PathEscape(key)}, opts, res, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return res, nil
+}
+
+// DecodeCipherText gets payload from vault ciphertext format (removes "vault:v<ver>:" prefix)
+func DecodeCipherText(vaultCipherText string) (string, error) {
```

Comment:
Return the `keyVersion` too. This allows us to re-encode a ciphertext with `EncodeCipherText` from the information returned by `DecodeCipherText`